### ISSUE TITLE
Table of Contents should warn about multiple H1

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#4230]
The post title may or may not be a H1 for the reasons explained in #4191. Ideally, there should be a way to detect what a theme is using for the post title, or the theme should declare it... hard to do, I now.

Regardless of the post title, it is possible to add one or multiple H1 headings in the post content. The table of contents should warn users that having multiple H1 in a page is a very bad practice and educate them to not do that.

![screen shot 2018-01-02 at 13 09 45](https://user-images.githubusercontent.com/1682452/34485992-c17f0cf2-efcd-11e7-9de1-c89f7a81247a.png)

**Comment by benoitchantre** [comment]
This is a great idea, and it would be a good thing to include bad nesting to to educate users when they don't use headings levels appropriately.

> To facilitate navigation and understanding of overall document structure, authors should use headings that are properly nested (e.g., h1 followed by h2, h2 followed by h2 or h3, h3 followed by h3 or h4, etc.).

source: W3C - [Organizing a page using headings](https://www.w3.org/TR/2012/NOTE-WCAG20-TECHS-20120103/G141)

**Comment by afercia** [comment]
@benoitchantre it already warns for incorrect hierarchy and empty headings, is that what you meant?

<img width="278" alt="screen shot 2018-01-03 at 09 03 39" src="https://user-images.githubusercontent.com/1682452/34512843-2b613784-f065-11e7-80ec-b3ef32ef19e4.png">

**Comment by benoitchantre** [comment]
@afercia yes, exactly. I hadn't noticed that before. Would it be possible to highlight the warning messages a little more?

Here's a suggestion:

<img width="278" alt="34512843-2b613784-f065-11e7-80ec-b3ef32ef19e4" src="https://user-images.githubusercontent.com/1521015/34537358-6cccc78e-f0c8-11e7-821a-7e8426be89fe.png">

Note: parenthesis in my example could be removed.

**Comment by afercia** [comment]
@benoitchantre that's more a question for designers and a bit out of the sco[e ot this issue. I'd just note the color should have a sufficient color contrast ratio with the background.
